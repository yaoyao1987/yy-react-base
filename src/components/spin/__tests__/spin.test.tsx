import React from 'react'
import { render } from '@testing-library/react'

import Spin, { SpinProps } from '../spin'

const testProps: SpinProps = {
  spinning: false,
  size: 'large',
}
describe('test Spin component', () => {
  it('sholud render correct component', () => {
    const wrapper = render(<Spin />)
    const element = wrapper.container.querySelector('.yy-spin') as HTMLElement
    expect(element.tagName).toEqual('SPAN')
    expect(element).toHaveClass('yy-spin yy-spin-spinning')
    expect(wrapper).toMatchSnapshot()
  })
  it('sholud render correct size', () => {
    const wrapper = render(<Spin {...testProps} />)
    const element = wrapper.container.querySelector('.yy-spin') as HTMLElement
    expect(element.tagName).toEqual('SPAN')
    expect(element).not.toHaveClass('yy-spin-spinning')
    expect(element).toHaveClass('yy-spin-large')
  })
})
