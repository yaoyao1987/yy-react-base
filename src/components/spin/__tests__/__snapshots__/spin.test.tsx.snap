// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Spin component sholud render correct component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <span
        class="yy-spin yy-spin-dot-spin yy-spin-spinning"
      >
        <i
          class="yy-spin-dot-item"
        />
        <i
          class="yy-spin-dot-item"
        />
        <i
          class="yy-spin-dot-item"
        />
        <i
          class="yy-spin-dot-item"
        />
      </span>
    </div>
  </body>,
  "container": <div>
    <span
      class="yy-spin yy-spin-dot-spin yy-spin-spinning"
    >
      <i
        class="yy-spin-dot-item"
      />
      <i
        class="yy-spin-dot-item"
      />
      <i
        class="yy-spin-dot-item"
      />
      <i
        class="yy-spin-dot-item"
      />
    </span>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
